src/js/service/printService.js

```diff
@@ -269,33 +294,106 @@ async function addGridToPdf(doc, gridData, options, metadata, globalGrid) {
     return Promise.all(promises);
 }
 
-function addLabelToPdf(doc, element, currentWidth, currentHeight, xStartPos, yStartPos, bgColor) {
+function addLabelToPdf(doc, element, currentWidth, currentHeight, xStartPos, yStartPos, bgColor, metadata, useElementColors = true) {
+    // 1. Get label and apply convert mode
     let label = i18nService.getTranslation(element.label);
+    let textConfig = metadata && metadata.textConfig ? metadata.textConfig : {};
     let hasImg = element.image && (element.image.data || element.image.url);
-    let fontSizeMM = hasImg ? currentHeight * (1 - pdfOptions.imgHeightPercentage) : currentHeight / 2;
-    let fontSizePt = (fontSizeMM / 0.352778) * 0.8;
-    let maxWidth = currentWidth - 2 * pdfOptions.textPadding;
+    let convertMode = textConfig.convertMode;
     if (convertMode === TextConfig.CONVERT_MODE_UPPERCASE) {
         label = label.toLocaleUpperCase();
     } else if (convertMode === TextConfig.CONVERT_MODE_LOWERCASE) {
         label = label.toLocaleLowerCase();
     }
-    let optimalFontSize = getOptimalFontsize(
-        doc,
-        label,
-        fontSizePt,
-        maxWidth,
-        currentHeight - 2 * pdfOptions.textPadding,
-        !hasImg
-    );
-    let textColor = fontUtil.getHighContrastColorRgb(bgColor);
-    doc.setTextColor(textColor[0], textColor[1], textColor[2]);
-    doc.setFontSize(optimalFontSize);
-    let dim = doc.getTextDimensions(label);
+    // 2. Font family (use only config/global value)
+    let fontFamily = textConfig.fontFamily || 'Arial';
+    try {
+        doc.setFont(fontFamily);
```

Comment:
I think this doesn't work like this. I think you have to load to font to the PDF before you can use it. See already existing private method `async function loadFont(path, doc)` around line `500`.